### ISSUE TITLE
Option ignore-decorators does not work properly

### ISSUE THREAD

**Issue by donpatrice** [PyCQA/pydocstyle#410]
The ignore-decorators option does not work, when the decorated method is the first method in  a class.

Consider the following example file named foo.py:
```
"""Shut up."""


class FooClass:
    """Shut up."""

    @property
    def bar(self):
        pass

```
Then
```
$ pydocstyle --ignore-decorators='.*' foo.py
foo.py:8 in public method `bar`:
        D102: Missing docstring in public method
```

While the example foo2.py
```
"""Shut up."""


class FooClass:
    """Shut up."""

    def foo(self):
        """Shut up."""
        pass

    @property
    def bar(self):
        pass
```
is working as expected:
```$ pydocstyle --ignore-decorators='.*' foo2.py```

**Comment by samj1912** [comment]
I was able to replicate this with something much more simpler

See
```python

@property
def test():
    pass
```

The above is not working as expected where as the samething without the newline in the begining will work.

```python
@property
def test():
    pass
```

**Comment by samj1912** [comment]
I was able to debug this, and fix the error in the parser that was causing pydocstyle to not recognize the decorators in some cases. Should be fixed by #411
